pkg/sync/sync.go

```diff
@@ -475,19 +557,25 @@ func producer(tasks chan<- object.Object, src, dst object.ObjectStorage, config
 		}
 
 		// FIXME: there is a race when source is modified during coping
-		if dstobj == nil || obj.Key() < dstobj.Key() || config.ForceUpdate ||
-			obj.Size() != dstobj.Size() || config.Update && obj.Mtime().After(dstobj.Mtime()) {
-			totalBytes += obj.Size()
-			copiedBytes.SetTotal(totalBytes, false)
+		if dstobj == nil || obj.Key() < dstobj.Key() {
 			tasks <- obj
-		} else { // obj.Key() == dstobj.Key()
-			if config.DeleteSrc && obj.Size() == dstobj.Size() {
+		} else { // obj.key == dstobj.key
+			if config.ForceUpdate ||
+				(config.Update && obj.Mtime().After(dstobj.Mtime())) ||
+				(!config.Update && obj.Size() != dstobj.Size()) {
+				tasks <- obj
+			} else if config.Update { // obj.mtime <= dstobj.mtime
```

Comment:
We should verify the checksum when obj.mtime == dstobj.mtime

Reply:
source and destination may have different time resolution, should we round it to seconds?